sql/plan/truncate.go

```diff
@@ -0,0 +1,170 @@
+package plan
+
+import (
+	"github.com/dolthub/go-mysql-server/sql"
+	"gopkg.in/src-d/go-errors.v1"
+)
+
+var ErrTruncateNotSupported = errors.NewKind("table doesn't support TRUNCATE")
+
+// Truncate is a node describing the deletion of all rows from some table.
+type Truncate struct {
+	db  string
+	tbl sql.Node
+}
+
+var _ sql.Node = (*Truncate)(nil)
+var _ sql.DebugStringer = (*Truncate)(nil)
+
+// NewTruncate creates a Truncate node.
+func NewTruncate(db string, table sql.Node) *Truncate {
+	return &Truncate{
+		db:  db,
+		tbl: table,
+	}
+}
+
+func GetTruncatable(node sql.Node) (sql.TruncateableTable, error) {
+	switch node := node.(type) {
+	case sql.TruncateableTable:
+		return node, nil
+	case *IndexedTableAccess:
+		return GetTruncatable(node.ResolvedTable)
+	case *ResolvedTable:
+		return getTruncatableTable(node.Table)
+	}
+	for _, child := range node.Children() {
```

Comment:
Anytime you're inspecting all children of a node like this, use plan.Inspect instead of doing it yourself

Reply:
I was just following the convention that all of the other nodes follow.